ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -652,6 +652,7 @@ struct vk_flash_attn_push_constants {
     uint32_t split_kv;
     uint32_t k_num;
 };
+static_assert(sizeof(vk_flash_attn_push_constants) <= 128, "sizeof(vk_flash_attn_push_constants) must be <= 128");
```

Comment:
Is this 128 byte a hard limit of the size of the struct? I need to add `nem3` for the change in https://github.com/ggml-org/llama.cpp/pull/14505 and it will cross the limit.

Reply:
128 is the minimum required by the Vulkan spec, although I think it went up to 256 with Vulkan 1.4. The Vulkan device database has a list: https://vulkan.gpuinfo.org/displaydevicelimit.php?name=maxPushConstantsSize&platform=all

If we want to support older implementations/drivers, it has to stay at or below 128 (or shaders that need more have to be disabled on devices that don't support them)

Reply:
128 bytes is the minimum required push constant size, so if we go over that then we would need to check for support and we would probably lose a lot of devices (see http://vulkan.gpuinfo.org/displaydevicelimit.php?name=maxPushConstantsSize&platform=all).

If you just need one more dword, there are some existing fields you could pack together, e.g. mask (which is just a bool) and maybe n_head_log2. Sadly, 16-bit push constants are not available everywhere so you'll probably need to pack them into a uint32_t.